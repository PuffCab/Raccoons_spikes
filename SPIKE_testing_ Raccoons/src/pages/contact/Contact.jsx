import React, { useState } from "react";

function Contact() {
  const [email, setEmail] = useState("");
  const [name, setName] = useState("");
  const [isVisible, setisVisible] = useState(false);
  const showSpan = () => {
    setTimeout(() => {
      setisVisible(true);
    }, 2000);
  };

  return (
    <>
      <h1>Contact</h1>
      <h1>Contact</h1>
      <span
        data-testid="span-text"
        style={{ visibility: isVisible ? "visible" : "hidden" }}
      >
        {`Hi ${name}, message sent to ${email}`}
      </span>
      <form action="">
        <input
          type="text"
          name="name"
          id="name"
          placeholder="Name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
        <input
          type="email"
          name="email"
          id="email"
          placeholder="Email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </form>
      <button onClick={showSpan}>Contact Me</button>
    </>
  );
}

export default Contact;
