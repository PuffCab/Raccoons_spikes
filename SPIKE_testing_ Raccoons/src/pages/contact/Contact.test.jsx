import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { expect, test } from "vitest";
import Contact from "./Contact";

test("header element renders", () => {
  render(<Contact />);

  const headerElement = screen.getByRole("heading");

  expect(headerElement).toBeInTheDocument();
  expect(headerElement.innerHTML).toBe("Contact");
});
test("header contains text Contact", () => {
  render(<Contact />);

  const headerElement = screen.getByRole("heading");

  expect(headerElement).toBeInTheDocument();
  expect(headerElement.innerHTML).toBe("Contact");
});
test("renders two imputs with placerholder value name and email", () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("Name");
  const emailInput = screen.getByPlaceholderText("Email");

  expect(nameInput, emailInput).toBeInTheDocument();
});
test("renders a button element", () => {
  render(<Contact />);

  const button = screen.getByRole("button");

  expect(button).toBeInTheDocument();
});
test("span shouldn't be initially visible", () => {
  render(<Contact />);

  //if we want to text the non existance of an element
  // const span = screen.queryByTestId("span-text");
  // expect(span).toBeNull();

  const span = screen.getByTestId("span-text");

  expect(span).not.toBeVisible();
});
test("inputs  should be initially empty", () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("Name");
  const emailInput = screen.getByPlaceholderText("Email");

  expect(nameInput.value).toBe("");
  expect(emailInput.value).toBe("");
});
test("inputs  should display a value when user types ", () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("Name");
  const emailInput = screen.getByPlaceholderText("Email");
  const nameInputText = "Faruk";
  const emailInputText = "faruk@email.com";

  //Fire event
  fireEvent.change(nameInput, { target: { value: nameInputText } });
  fireEvent.change(emailInput, { target: { value: emailInputText } });

  expect(nameInput.value).toBe(nameInputText);
  expect(emailInput.value).toBe(emailInputText);
});

test("span  should display user name and email after button click ", async () => {
  render(<Contact />);

  const span = screen.getByTestId("span-text");
  const button = screen.getByRole("button");
  const nameInput = screen.getByPlaceholderText("Name");
  const emailInput = screen.getByPlaceholderText("Email");
  const nameInputText = "Faruk";
  const emailInputText = "faruk@email.com";

  //Fire event
  fireEvent.change(nameInput, { target: { value: nameInputText } });
  fireEvent.change(emailInput, { target: { value: emailInputText } });
  fireEvent.click(button);

  await waitFor(
    () => {
      expect(span).toBeVisible();
      expect(span).toHaveTextContent(
        `Hi ${nameInputText}, message sent to ${emailInputText}`
      );
    },
    { timeout: 2000 }
  );
});

test("Contact component is the same", () => {
  const { container } = render(<Contact />);
  // console.log("element :>> ", element);
  expect(container).toMatchSnapshot();
});
