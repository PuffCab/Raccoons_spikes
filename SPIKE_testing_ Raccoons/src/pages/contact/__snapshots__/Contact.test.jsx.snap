// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact component is the same 1`] = `
<div>
  <h1>
    Contact
  </h1>
  <span
    data-testid="span-text"
    style="visibility: hidden;"
  >
    Hi , message sent to 
  </span>
  <form
    action=""
  >
    <input
      id="name"
      name="name"
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      id="email"
      name="email"
      placeholder="Email"
      type="email"
      value=""
    />
  </form>
  <button>
    Contact Me
  </button>
</div>
`;
